protocol/communities/persistence.go

```diff
@@ -0,0 +1,131 @@
+package communities
+
+import (
+	"crypto/ecdsa"
+	"database/sql"
+
+	"github.com/golang/protobuf/proto"
+	"go.uber.org/zap"
+
+	"github.com/status-im/status-go/eth-node/crypto"
+	"github.com/status-im/status-go/protocol/protobuf"
+)
+
+type Persistence struct {
+	db     *sql.DB
+	logger *zap.Logger
+}
+
+func (p *Persistence) SaveCommunity(community *Community) error {
+	id := community.ID()
+	privateKey := community.PrivateKey()
+	description, err := community.ToBytes()
+	if err != nil {
+		return err
+	}
+
+	_, err = p.db.Exec(`INSERT INTO communities_communities (id, private_key, description, joined, verified) VALUES (?, ?, ?,?,?)`, id, crypto.FromECDSA(privateKey), description, community.config.Joined, community.config.Verified)
+	return err
+}
+
+func (p *Persistence) queryCommunities(query string) ([]*Community, error) {
+	var response []*Community
+
+	rows, err := p.db.Query(query)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
```

Comment:
A tiny little thing, `rows.Close()` returns an error, so if anything goes wrong with closing the `rows` the error will never be handled.

If we care about this, something like below would work to catch the error.

```suggestion
func (p *Persistence) queryCommunities(query string) (response []*Community, err error) {
	rows, err := p.db.Query(query)
	if err != nil {
		return nil, err
	}
	defer func() {
		err = rows.Close()
	}()

```